pages/index.tsx

```diff
@@ -1,72 +1,88 @@
-import type { NextPage } from 'next'
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import type { NextPage } from 'next';
+import Head from 'next/head';
+import { useEffect, useState } from 'react';
+import Endpoints from '../src/config/Endpoints';
+import { Schedule } from '../src/domain/models/apiSchema';
+import ErrorAlert from '../src/presentation/elements/error';
+import Loader from '../src/presentation/elements/loader';
+import HomePageLayout from '../src/presentation/pages/home';
 
 const Home: NextPage = () => {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+  const [schedule, setSchedule] = useState<Array<Schedule>>([]);
+  const [countryCode, setCountryCode] = useState("");
+  const [countryCodeError, setCountryCodeError] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [errorWhileCatchingShows, setErrorWhileCatchingShows] = useState(false);
 
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+  useEffect(() => {
+    if (countryCodeError) return;
+    getUserCountryCode();
+  }, [countryCodeError])
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.tsx</code>
-        </p>
+  useEffect(() => {
+    if (errorWhileCatchingShows) return;
+    if (!countryCodeError && countryCode && schedule.length < 1) {
+      getAllShows(`${Endpoints.tvScheduleEndpoint}?country=${countryCode}`);
+    }
+    if (countryCodeError) {
+      getAllShows(Endpoints.tvScheduleEndpoint);
+    }
+  })
```

Comment:
You usually will want to ensure you have populated your dependency array with the props it's consuming otherwise this effect may re-execute on every render unnecessarily. 

```suggestion
  useEffect(() => {
    if (errorWhileCatchingShows) return;
    if (!countryCodeError && countryCode && schedule.length < 1) {
      getAllShows(`${Endpoints.tvScheduleEndpoint}?country=${countryCode}`);
    }
    if (countryCodeError) {
      getAllShows(Endpoints.tvScheduleEndpoint);
    }
  }, [countryCode, countryCodeError, errorWhileCatchingShows, schedule.length]);
```